be2-scala/linter/rules/src/main/scala/fix/NanComparison.scala

```diff
@@ -0,0 +1,46 @@
+/*
+rule = NanComparison
+*/
+package fix
+
+import scalafix.lint.LintSeverity
+import scalafix.v1._
+import scala.meta._
+
+case class NanComparisonDiag(tree: Tree) extends Diagnostic {
+  override def message: String = "Comparison with NaN"
+
+  override def severity: LintSeverity = LintSeverity.Error
+
+  override def explanation: String = "NaN comparison will always fail. Use value.isNaN instead."
+
+  override def position: Position = tree.pos
+}
+
+class NanComparison extends SemanticRule("NanComparison") {
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+
+    def matcher(l: Any, r: Any, t: Tree): Patch = {
+      // Either we have definitions to extract or we have direct terms and matching is the same
+      // => matcher function avoids repetition. Type of arguments is any since findDefinition will return Any,
+      // but it doesn't matter since we do pattern matching
+      (l, r) match {
+        case (Term.Select(Term.Name("Double"), Term.Name("NaN")), _) | (_, Term.Select(Term.Name("Double"), Term.Name("NaN"))) => Patch.lint(NanComparisonDiag(t))
+        case _ => Patch.empty
+      }
+    }
+
+    doc.tree.collect {
+      case t @ Term.ApplyInfix.After_4_6_0(lhs, Term.Name("=="), _, Term.ArgClause(List(rhs), _)) =>
```

Comment:
Like last time, shouldn't .equals be also handled ?

Reply:
Yes! Added it with my last commit.
You'll notice I didn't test for .equals between Double and Float and Float and Double because it already throws a warning upon compilation (comparing values of types Float and Double using `equals` unsafely bypasses cooperative equality; use `==` instead).